src/libraries/System.Text.Json/src/System/Text/Json/ThrowHelper.Serialization.cs

```diff
@@ -243,22 +243,20 @@ public static void ThrowInvalidOperationException_IgnoreConditionOnValueTypeInva
         [MethodImpl(MethodImplOptions.NoInlining)]
         public static void ThrowInvalidOperationException_NumberHandlingOnPropertyInvalid(JsonPropertyInfo jsonPropertyInfo)
         {
-            MemberInfo? memberInfo = jsonPropertyInfo.MemberInfo;
-
-            if (!jsonPropertyInfo.ConverterBase.IsInternalConverter)
+            if (jsonPropertyInfo.IsForClassInfo)
```

Comment:
Here, I simplified the exceptions thrown when we detect invalid number handling settings because it would be complex to fetch additional information about the converter for the collection element (if applicable).